### PR TITLE
Add feature to accomodate missing conjunctions

### PR DESCRIPTION
**Pull request by li-kai**
Well sadly CI not set up to run tests for api yet.

### CODE REVIEW

File: api/gulp-tasks/local/genReqTree/parseString.js [hunk 1]
```diff
@@ -215,7 +245,8 @@ function parseString(pre, log) {
   const parser = new ReqTreeParser(tokens);
   const result = parser.parse();
   if (parser.errors.length > 0) {
-    log.error(`'${pre}' encoutered parsing errors:\n${parser.errors}`);
+    console.log(result, parser.errors);
```
Comment (ZhangYiJiang) [inline api/gulp-tasks/local/genReqTree/parseString.js]: So, why log instead of error?

File: api/gulp-tasks/local/genReqTree/parseString.js [hunk 2]
```diff
@@ -175,7 +180,32 @@ function cleanOperators(tokens) {
...
+  const singularConjunction = removedDuplicates.reduce((acc, token) => {
+    const { image } = token;
+    if (image === 'and' || image === 'or') {
+      if (acc === null) return token;
+      if (image !== acc.image) return false;
+    }
+    return acc;
+  }, null);
+  // Fill in missing values with conjunction found
+  for (let i = 0; i < removedDuplicates.length; i += 1) {
```
Comment (ZhangYiJiang) [inline api/gulp-tasks/local/genReqTree/parseString.js]: Add `allowForLoopAfterthoughts` and change this to `i++`, so it match the config in the other projects?

File: api/gulp-tasks/local/genReqTree/parseString.js [hunk 3]
```diff
@@ -175,7 +180,32 @@ function cleanOperators(tokens) {
     // then check if each element is different than the one before it
     return !(AND_OR_REGEX.test(currentImage) && AND_OR_REGEX.test(nextImage));
   });
-  return removedDuplicates;
+
+  const moduleTokens = [];
+  // Falsy value if array does not contain unique conjunction
+  const singularConjunction = removedDuplicates.reduce((acc, token) => {
```
Comment (ZhangYiJiang) [inline api/gulp-tasks/local/genReqTree/parseString.js]: Would it be more readable if you count the number of `and` and `or` make sure it's equal to 1?

Comment (ZhangYiJiang) [reply api/gulp-tasks/local/genReqTree/parseString.js]: Also best to add a comment explaining why this is necessary?

Comment (li-kai) [reply api/gulp-tasks/local/genReqTree/parseString.js]: We need the token itself so counting wouldn't really make it more readable.

**Pull request merged**